test/DynamoCoreTests/DSCoreDataTests.cs

```diff
@@ -186,5 +191,359 @@ public void ParsingJSONInPythonReturnsSameResult()
             // Verify values match when parsing JSON via Python
             AssertPreviewValue("cdad5bf1-f5f7-47f4-a119-ad42e5084cfa", true);
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForBoundingBoxReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Abstract_BoundingBox_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("abb39e07-db08-45cf-9438-478defffbf68", true);
+
+            // Verify current failure cases are all false
+            AssertPreviewValue("eb9130a1-309c-492a-9679-28ad5ef8fddf", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForCoordinateSystemReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Abstract_CoordinateSystem_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("07366adaf0954529b1ed39b240192c96", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForPlaneReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Abstract_Plane_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("9754cbd66d4842419a6899f372a80aee", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForVectorReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Abstract_Vector_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("71efc8c5c0c74189901707c30e6d5903", true);
+        }
+
+        [Test]
+        [Category("Failure")]
```

Comment:
why does this fail

Reply:
Failures are waiting for the upstream PR in LibG